cmd/xl-v1-healing.go

```diff
@@ -192,16 +193,18 @@ func shouldHealObjectOnDisk(xlErr, dataErr error, meta xlMetaV1, quorumModTime t
 	}
 	if xlErr == nil {
 		// If xl.json was read fine but there is some problem with the part.N files.
-		if dataErr == errFileNotFound {
+		if IsErr(dataErr, []error{
+			errFileNotFound,
+			errFileCorrupt,
+		}...) {
 			return true
 		}
-		if dataErr == errFileUnexpectedSize {
+		// Usually this error is not typed and we can't rely on it to be typed
+		// - so we have to do strings based comparison.
+		if dataErr != nil && strings.Contains(dataErr.Error(), "Bitrot verification mismatch") {
```

Comment:
@harshavardhana can you add new function IsBitrotVerificationMismatch() in cmd/storage-errors.go to check if the error string is a bitrot error? if we change the bitrot error message, all the healing will be non functional without knowning it (we don't have any functional tests for distributed healing yet)

Reply:
I will try to fix it by bringing in proper types..